src/session.cpp

```diff
@@ -281,7 +296,12 @@ class Impl : public dap::Session {
     } else {
       handlers.error("Unknown message type '%s'", type.c_str());
     }
-
+    } catch (std::exception& e) {
```

Comment:
[Google's C++ style does not use exceptions](https://google.github.io/styleguide/cppguide.html#Exceptions), and so I've been trying to avoid relying on these in cppdap.

The nlohmann JSON library has [a flag to disable the use of exceptions](https://github.com/nlohmann/json#further-notes), replacing code that raises them with calls to `abort()`. I've intentionally tried to isolate the The nlohmann JSON library with a single cpp file `json_serializer.cpp`. In here I've attempted to catch the cases that would trigger an exception ([example](https://github.com/google/cppdap/blob/master/src/json_serializer.cpp#L60-L62)), but clearly I've missed some cases.

If you're able to provide me a message payload that triggers an JSON execption, I can write a test and try to catch the case in `json_serializer.cpp`.

You have highlighted to me that I should probably invest some time into fuzzing the cppdap library. We can then feel confident that the library is robust to all kinds of payload.